How payjoin saves (#69)
payjoin/payjoin.org: docs/how-payjoin-saves.md

---
sidebar_position: 1
---

# How Can Payjoin Save 1/3 the Cost of Transacting?

Payjoin can save more than 1/3 of the cost of making normal transactions by batching them together.

Payjoin lets a sender and receiver combine their transactions when they use Bitcoin. This doesn't just save money. It fixes the biggest privacy problem the Bitcoin network has at the same time.

Imagine Alice would like to send some sats to Bob. Before Payjoin, she'd make a transaction that looks like this:

```
Alice 500,000 sats  ->  200,000 sats to Bob
                        292,387 sats to Alice

(7,613 sats are fees for 152.25 vB at 50 sat/vB)

> ```

Author: Let's find numbers that we're happy with before spending time on nice graphics. Content first.

> ```

Reviewer: note from convo: let's do small numbers in btc (Alice 1 btc, Bob 3 btc, etc)

So Alice makes one new UTXO to Bob, and another that holds the change to herself, since UTXOs must be spent completely. The difference between the inputs and outputs is paid to the miners as fees.

Before he gets paid by Alice, Bob knows he wants to consolidate the bitcoin from Alice with other bitcoin to Charlie, perhaps to exchange for local currency.

```
Bob 200,000 sats ->  491,575 sats to Charlie
Bob 300,000 sats 

(8,425 sats are fees for 168.5 vB at 50 sat/vB)
```

Space in the blockchain is limited, and transactions pay for base costs of the transaction header, costs for each input and costs for each output. Making these two transactions costs:

*2 × base costs + 3 × input costs + 3 × output costs*

If Bob knows he needs to pay Charlie, he can Payjoin with Alice to save loads of fees.

Their Payjoin looks like this

```
Alice 500,000 sats ->  500,000 sats to Charlie
Bob   300,000 sats     289,412 sats to Alice

(10,588 sats are fees for 211.25 vB at 50 sat/vB)
```

> Since Alice's payment to Bob "[cut-through](https://bitcointalk.org/index.php?topic=281848.0)" to Charlie,

Reviewer: Explanation that helped me understand: https://payjoin.org/docs/why-payjoin/scaling#transaction-cut-through

Take a fictional exchange with 5 BTC in their treasury selling 1 bitcoin each to Alice, bob, and Carol for example.

```
Exchange 5 btc  ->  1 btc to Alice
                   ~4 btc minus fees to Exchange
```

```
Exchange 4 btc  ->  1 btc to Bob
                   ~3 btc minus fees to Exchange
```

```
Exchange 3 btc  ->  1 btc to Carol
                   ~2 btc minus fees to Exchange
```

Each transaction would cost the exchange 𝑏 + 𝑖 + 2𝑜, and they would pass the fees onto their customers in order to make a profit. The sum of these costs would be 3𝑏 + 3𝑖 + 6𝑜, which would come out of the final ~2 BTC change the exchange keeps in the end.

## Old-school Payment Batching

Batching helps the exchange save time and money in two ways. First, the overall cost to post such a transaction is cheaper than the cost of making three individual transactions to produce the same result. Second, a single unspent output can fund multiple withdrawals without waiting for each one to settle.

> your typical exchange withdrawal looks like this:

Reviewer: ```suggestion
Your typical exchange withdrawal looks like this:
```

```
Exchange 5 btc  ->   1 btc to Alice
                     1 btc to Bob
                     1 btc to Carol
                    ~2 btc minus fees to Exchange
```

All else being equal, an exchange making this batch instead of three separate transactions will only pay 𝑏 + 𝑖 + 4𝑜 in fees, saving 2𝑏 + 2𝑖 + 2𝑜 compared to 3𝑏 + 3𝑖 + 6𝑜, at least 33% cheaper than making those three transactions separately. Second, the exchange does not have to wait for each withdrawal to settle before paying out the next one since they can service them all with a single UTXO. Sure, the exchange could keep multiple UTXOs ready for spending, but that is always going come at the cost of making even more transactions in preparation.

## Payjoin Payment Batching

> What if Dave the depositor can payjoin? He gets a benefit of breaking some privacy heuristics, and he can save the exchange some fees that might be able to be pased on to him. Payjoin lets the exchange fund withdrawals with Dave's deposit in the same transaction.

Reviewer: ```suggestion
What if Dave the depositor can payjoin? He gets a benefit of breaking privacy heuristics, and he can save the exchange some fees that could be passed on to him. Payjoin lets the exchange fund withdrawals with Dave's deposit in the same transaction.
```

Dave knows that Payjoin was used, but not which outputs are withdrawals vs consolidations.
The Exchange can see everything as it could before, but an outside observer cannot without that information being leaked.
Alice, Bob, and Carol can't tell whether or not all inputs came from the exchange or include an outsider's deposit. They know they got paid and that's their main concern.

In total, the payjoin only cost

*1 × base costs + 2 × input costs + 2 × output costs*

saving at least 1/3 of the cost of a regular transaction, and fees were split between Dave and the Exchange by custom.[^1]

## Future Payjoin Batching

Bitcoin technically allows even more depositors to batch their transactions, too. Even though today's version of Payjoin only allows Dave to pay the exchange, future versions of Payjoin will allow others pay the exchange and each other in the same transaction too. More batching means more saving, and potentially more privacy too.

```
Dave     3 btc  ->  1 btc to Alice
Erin     1 btc      1 btc to Bob
Frank    2 btc      1 btc to Carol
Exchange 2 btc     ~5 btc minus fees to Exchange
```

> However, such transactions are more difficult to coordinate, so it will take an effort to develop this new protocol and get it deployed. Integrating Payjoin V2 can save money, improve privacy, and help get to the next iteration that massively fixes bitcoin's privacy through batching. Batching Bitcoin, saving sats, and preserving privacy seem like different goals, but with a little communication, the three goals can be achieved in every single transaction.

Reviewer: ```suggestion
However, such transactions are more difficult to coordinate, so it will take an effort to develop this new protocol and get it deployed. Integrating Payjoin V2 can save money, improve privacy, and help get to the next iteration that massively fixes Bitcoin's privacy through batching. Batching bitcoin, saving sats, and preserving privacy seem like different goals, but with a little communication, they can be achieved in every single transaction.
```
